src/v/kafka/server/group.cc

```diff
@@ -1176,6 +1176,38 @@ void group::fail_offset_commit(
 
 void group::reset_tx_state(model::term_id term) { _term = term; }
 
+ss::future<txn_offset_commit_response>
+group::store_txn_offsets(txn_offset_commit_request&& req) {
+    if (_partition->term() != _term) {
+        co_return txn_offset_commit_response(
+          req, error_code::unknown_server_error);
```

Comment:
wouldn't this be `term_missmatch_error` ?

Reply:
This is true, but I don't see value in introducing a precise error code without creating a handler specialized for that error. And  I have a couple tickets which will address this case in the future e.g. https://app.clubhouse.io/vectorized/story/2187 but so far unknown_server_error does the job